### PR TITLE
Updated code according to latest database modelling

### LINKED ISSUE #32: [BUG] Apply changes according to the latest database modelling
**Describe the bug**
- Modify `CreateTable.js` script as per the latest database modelling.
- Update `model.md` as per the new table design.
- Check/Update query operations according to the latest schema for all lambdas.

Area Where Bug Occurred
- [X] Lambda 
- [X] DynamodDb

**Before submitting an issue check the below**
- [X] I have synced all my node versions as mentioned in the project"
- [X] I am using the same version of npm as is the project

### PR DESCRIPTION
**Pull request by lakshya-20**
- Modified `CreateTable.js` script as per the latest database modelling.
- Updated `model.md` as per the new table design.
- Updated lambdas.

### CODE REVIEW

File: DynamoDB/CreateTable.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
 var AWS = require("aws-sdk");
 
 /*Use this for AWS Console*/
-//AWS.config.loadFromPath('../keys.json');
+AWS.config.loadFromPath('../keys.json');
```
Comment (ayush0x00) [inline DynamoDB/CreateTable.js]: comment out the codes that are used while local testing

File: lambdas/file_access_patterns/lambda_for_fileDelete/index.js [hunk 2]
```diff
@@ -16,31 +16,34 @@ function response(statusCode, message) {
 }
 
 exports.deleteFile = async (event) => {
-  const reqBody = event.body;
+  const reqBody = JSON.parse(event.body);
```
Comment (ayush0x00) [inline lambdas/file_access_patterns/lambda_for_fileDelete/index.js]: don't need to parse..the api gateway is configured to parse and send the event.

File: lambdas/file_access_patterns/lambda_for_fileUpload/index.js [hunk 3]
```diff
@@ -17,18 +13,18 @@ function response(statusCode, message) {
 
 exports.uploadFile = async (event) => {
   // TODO implement
-  const reqBody = event.body;
+  const reqBody = JSON.parse(event.body);
```
Comment (ayush0x00) [inline lambdas/file_access_patterns/lambda_for_fileUpload/index.js]: same here..no need of parsing the event

File: lambdas/user_access_patterns/lambda_for_createUser/index.js [hunk 4]
```diff
@@ -1,11 +1,6 @@
 const AWS = require("aws-sdk");
 
 //AWS.config.loadFromPath("../../../keys.json");
```
Comment (ayush0x00) [inline lambdas/user_access_patterns/lambda_for_createUser/index.js:3]: I think its better to comment out the local setup code

File: DynamoDB/CreateTable.js (hunk 1 above)
Comment (danishjamal104) [reply DynamoDB/CreateTable.js]: agreed

File: lambdas/file_access_patterns/lambda_for_fileDelete/index.js (hunk 2 above)
Comment (danishjamal104) [reply lambdas/file_access_patterns/lambda_for_fileDelete/index.js]: we are using integration mapping template so no need to do this.

File: lambdas/file_access_patterns/lambda_for_fileDelete/index.js [hunk 5]
```diff
@@ -16,31 +16,34 @@ function response(statusCode, message) {
...
   var params = {
     TableName: "V-Transfer",
     Key: {
-      PK: `USER#${reqBody.user_email}`,
-      SK: `#FILE#${reqBody.timestamp}`,
+      PK: `USER#${reqBody.u_email}`,
+      SK: `#FILE#${reqBody.f_timestamp}`,
```
Comment (danishjamal104) [inline lambdas/file_access_patterns/lambda_for_fileDelete/index.js]: avoid using suffixes for variable names, rather use contextual attribute names. 
This lambda is for file deletion so, why we are taking file timestamps from the body instead make use of wildcard endpoints and utilize path variables.
`https://domain/file/{file_timestamp}`, then access it in lambda by `event.path.file_timestamp` won't that make more sense??

File: lambdas/file_access_patterns/lambda_for_fileDelete/index.js [hunk 6]
```diff
@@ -16,31 +16,34 @@ function response(statusCode, message) {
...
-  return response(201,"File delete success")
-}
+      TableName: "V-Transfer",
+      Key: {
+        PK: `USER#${reqBody.u_email}`,
+        SK: `METADATA`, 
+      },
+      UpdateExpression: "add storage_used :size",
+      ExpressionAttributeValues: {
+        ":size": -reqBody.f_size,
```
Comment (danishjamal104) [inline lambdas/file_access_patterns/lambda_for_fileDelete/index.js]: why the user will send the file size, utilize the attribute`ReturnValues: NONE | ALL_OLD | UPDATED_OLD | ALL_NEW | UPDATED_NEW`, to get back the data of deleted file on line 29, and access file size from their

File: lambdas/file_access_patterns/lambda_for_fileUpload/index.js [hunk 7]
```diff
@@ -1,12 +1,8 @@
 const AWS = require("aws-sdk");
-const crypto = require("crypto");
-const dynamo = new AWS.DynamoDB.DocumentClient();
+
```
Comment (danishjamal104) [inline lambdas/file_access_patterns/lambda_for_fileUpload/index.js]: avoid unnecessary changes, like adding empty lines.

File: lambdas/file_access_patterns/lambda_for_fileUpload/index.js [hunk 8]
```diff
@@ -1,12 +1,8 @@
...
 //AWS.config.loadFromPath("../../../keys.json");
-//Use the below code for local setup
-/*AWS.config.update({
-  region: "local",
-  endpoint: "http://localhost:8000",
-});*/
+
```
Comment (danishjamal104) [inline lambdas/file_access_patterns/lambda_for_fileUpload/index.js]: here too

File: lambdas/file_access_patterns/lambda_for_fileUpload/index.js (hunk 3 above)
Comment (danishjamal104) [reply lambdas/file_access_patterns/lambda_for_fileUpload/index.js]: agreed

File: lambdas/file_access_patterns/lambda_for_fileUpload/index.js [hunk 9]
```diff
@@ -17,18 +13,18 @@ function response(statusCode, message) {
...
   // TODO implement
-  const reqBody = event.body;
+  const reqBody = JSON.parse(event.body);
   var d = new Date();
   const timestamp = d.getFullYear()  + "-" + (d.getMonth()+1) + "-" + d.getDate() + "-" +
     d.getHours() + "-" + d.getMinutes()+ "-" +d.getSeconds();
 
   const file = {
-    PK: `USER#${reqBody.user_email}`,
+    PK: `USER#${reqBody.u_email}`,
```
Comment (danishjamal104) [inline lambdas/file_access_patterns/lambda_for_fileUpload/index.js]: please avoid using suffixes for attribute names.
Consider the URL path variable

File: lambdas/file_access_patterns/lambda_for_fileUpload/index.js [hunk 10]
```diff
@@ -38,18 +34,18 @@ exports.uploadFile = async (event) => {
...
-      SK: `METADATA`,  //token is password hash
-    },
-    UpdateExpression: "add storage_used :file_size",
-    ExpressionAttributeValues: {
-      ":file_size": reqBody.size,
-    },
-  }).promise()
-  return response(201,"File upload success")
-}
+      TableName: "V-Transfer",
```
Comment (danishjamal104) [inline lambdas/file_access_patterns/lambda_for_fileUpload/index.js:39]: use separate try/catch for each dynamo call
```
try{ dynamo.put().promise }
catch {//handle put error here }
try { dynamo.update().promise() }
catch{ //handle update error here }
```

File: lambdas/user_access_patterns/lambda_for_createUser/index.js (hunk 4 above)
Comment (danishjamal104) [reply lambdas/user_access_patterns/lambda_for_createUser/index.js:3]: no need to remove comments, we are suppose to push production code not the testing code.

File: lambdas/user_access_patterns/lambda_for_createUser/index.js [hunk 11]
```diff
@@ -15,26 +10,24 @@ function response(statusCode, message) {
   };
 }
 
-
 module.exports.createUser = async (event) => {
-  const reqBody = event.body;
+  let reqBody = JSON.parse(event.body);
   const user = {    
-    PK: `USER#${reqBody.user_email}`,
+    PK: `USER#${reqBody.u_email}`,
```
Comment (danishjamal104) [inline lambdas/user_access_patterns/lambda_for_createUser/index.js]: avoid using attributes suffix.

**Comment by danishjamal104** [comment]
Also @ayush0x00 and @lakshya-20 , design lambda corresponding to the endpoints this will make the task easy.

File: lambdas/file_access_patterns/lambda_for_fileDelete/index.js (hunk 2 above)
Comment (lakshya-20) [reply lambdas/file_access_patterns/lambda_for_fileDelete/index.js]: Okay

File: lambdas/file_access_patterns/lambda_for_fileUpload/index.js (hunk 3 above)
Comment (lakshya-20) [reply lambdas/file_access_patterns/lambda_for_fileUpload/index.js]: okay

File: lambdas/user_access_patterns/lambda_for_createUser/index.js (hunk 4 above)
Comment (lakshya-20) [reply lambdas/user_access_patterns/lambda_for_createUser/index.js:3]: > no need to remove comments, we are suppose to push production code not the testing code.

Agreed

File: lambdas/file_access_patterns/lambda_for_fileDelete/index.js (hunk 5 above)
Comment (lakshya-20) [reply lambdas/file_access_patterns/lambda_for_fileDelete/index.js]: As decided we are taking `u_id` as a path variable so I think `f_timestamp` should be taken in the body object. What's say?

File: lambdas/file_access_patterns/lambda_for_fileUpload/index.js (hunk 10 above)
Comment (lakshya-20) [reply lambdas/file_access_patterns/lambda_for_fileUpload/index.js:39]: Okay

File: lambdas/file_access_patterns/lambda_for_fileDelete/index.js (hunk 6 above)
Comment (lakshya-20) [reply lambdas/file_access_patterns/lambda_for_fileDelete/index.js]: Okay, this looks more efficient. Thanks

File: lambdas/file_access_patterns/lambda_for_fileUpload/index.js (hunk 9 above)
Comment (lakshya-20) [reply lambdas/file_access_patterns/lambda_for_fileUpload/index.js]: Okay

**Comment by lakshya-20** [comment]
@ayush0x00 @danishjamal104  I have done the changes required please have a look :)

File: lambdas/file_access_patterns/lambda_for_fileUpload/index.js (hunk 7 above)
Comment (danishjamal104) [reply lambdas/file_access_patterns/lambda_for_fileUpload/index.js]: remove this

File: lambdas/file_access_patterns/lambda_for_fileUpload/index.js (hunk 8 above)
Comment (danishjamal104) [reply lambdas/file_access_patterns/lambda_for_fileUpload/index.js]: this too

**Review by danishjamal104 — CHANGES_REQUESTED** [review]
Remove the unnecessary changes , and we are good to go

**Review by danishjamal104 — APPROVED** [review]
Everything looks good

File: lambdas/file_access_patterns/lambda_for_fileDelete/index.js [hunk 12]
```diff
@@ -20,27 +19,36 @@ exports.deleteFile = async (event) => {
...
-  return response(201,"File delete success")
-}
+      TableName: "V-Transfer",
+      Key: {
+        PK: `USER#${event.path.u_id}`,
+        SK: `METADATA`, 
+      },
+      UpdateExpression: "add storage_used :size",
+      ExpressionAttributeValues: {
+        ":size": -fileData.Attributes.size,
```
Comment (ayush0x00) [inline lambdas/file_access_patterns/lambda_for_fileDelete/index.js:44]: Is the file size being decremented from user storage_used properly?

Comment (lakshya-20) [reply lambdas/file_access_patterns/lambda_for_fileDelete/index.js:44]: Yes
